src/Package/Impl/DataInspect/Viewers/ObjectDetailsViewerProvider.cs

```diff
@@ -28,8 +29,8 @@ public sealed class ObjectDetailsViewerProvider : IObjectViewer {
             }
         }
 
-        public async Task ViewFile(string fileName, string tabName, bool deleteFile) {
-            await VsAppShell.Current.SwitchToMainThreadAsync();
+        public async Task ViewFile(string fileName, string tabName, bool deleteFile, CancellationToken cancellationToken) {
+            await VsAppShell.Current.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
Yeah, that's probably a good idea to verify all the usages of `SwitchToMainThreadAsync`. I've made this change because I've been checking all the methods that are called from `RHost` message loop for possible UI deadlocks. #2825